docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,390 +1,565 @@
--   [Gradient](#gradientvisual)
--   [Image](#imagevisual)
--   [Mesh](#meshvisual)
--   [Primitive](#primitivevisual)
--   [Text](#textvisual)
--   [Wireframe](#wireframevisual)
+Follow the steps in this section to familiarize yourself with visuals:
+- [how to create a Visual](#visualCreation)
+- [how to use a VisualView](#visualView)
+- [common properties for all visuals - VisualMap](#visualMap)
+- [visual types available in NUI with examples](#visualTypes) 
 
-When you use or create a visual, you must always specify the visual type through a property map. The visual type is required to avoid ambiguity as multiple visuals can be capable of rendering the same content.
 
-<a name="visualcreation"></a>
-## Creating and Registering Visuals
+ 
+<!---------------------------------------------------------------------------------------------------------------------------->
+<a name="visualCreation"></a>
+## Create visual
 
-To create a visual:
+The creation process of a visual is shown using a `BorderVisual` as an example.
+The following steps are required: 
 
-1.  Create a property map to set the visual properties.
+1.  Create a visual object of a desirable type:
 
-    You can use property maps in two ways:
+    ```csharp
+    BorderVisual _borderVisual = new BorderVisual();
+    ```
 
-    -   Use a specific `xxxProperty` structure for the visual, such as [Tizen.NUI.ColorVisualProperty](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.ColorVisualProperty.html), which specifies the properties for that visual type.
-    -   Use the `xxxVisual` [visual maps](#visualmap), such as `ColorVisual`.
+2.  Specify the properties required for the visual type. The following section provides detailed information on which properties are mandatory for a given visual:
 
-2.  Add other required property values to the property map.
+    ```csharp
+    // Set the border thickness
+    _borderVisual.BorderSize = 10.0f;
+    // The border color
+    _borderVisual.Color = Color.Blue;
+    ```
 
-    > **Note**  
-    > Add the visual type to the property map as the first entry.
+3.  You can specify the optional properties of a given visual, and also specify the properties that are common for all visual types:
+
+    ```csharp
+    // The size of a visual given as a percentage of the size of the parent
+    _borderVisual.RelativeSize = new RelativeVector2(0.5f, 0.5f);
+    // The parent's reference point
+    _borderVisual.Origin = Visual.AlignType.TopBegin;
+    // The Visual's reference point
+    _borderVisual.AnchorPoint = Visual.AlignType.TopBegin;
+    // Shift between Origin and AnchorPoint given as a percentage of a parent size 
+    _borderVisual.RelativePosition = new RelativeVector2(0.25f, 0.25f);
+    ```
 
-3.  Create the visual in a *factory* using the property map.
+4.  To add the visual to an already existing `VisualView` (more details regarding `VisualView` available in the [next section](#visualView)) use the `AddVisual(...)` method:
```

Comment:
```suggestion
4.  To add the visual to an already existing `VisualView` use the `AddVisual()` method:
```

- The "Use VisualView" section is right after this, so I think "(more details regarding `VisualView` available in the [next section](#visualView))" can be skipped. This section is "Create visual", so we can focus on "creating".

- Usually we write functions as "AddVisual()", so please remove the dots in ().